UnitsNet.Tests/QuantityComparisonTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using UnitsNet.Units;
+using Xunit;
+
+namespace UnitsNet.Tests
+{
+    public class QuantityComparisonTests
+    {
+        [Fact]
+        public void CompareTo_WithSameQuantityInAnotherUnit_ReturnsZero()
+        {
+            var firstMass = Mass.FromGrams(0.001);
+            var secondMass = firstMass.ToUnit(MassUnit.Microgram);
+
+            Assert.Equal(0, firstMass.CompareTo(secondMass));
+            Assert.Equal(0, secondMass.CompareTo(firstMass));
+        }
+
+        [Fact]
+        public void CompareTo_WithVeryCloseQuantity_ReturnsNonZero()
+        {
+            var firstMass = Mass.FromGrams(0);
+            var secondMass = Mass.FromGrams(double.Epsilon);
+
+            Assert.Equal(-1, firstMass.CompareTo(secondMass));
+            Assert.Equal(1, secondMass.CompareTo(firstMass));
+        }
+
+        [Fact]
+        public void Equals_WithSameQuantityInAnotherUnit_ReturnsTrue()
+        {
+            var firstMass = Mass.FromGrams(0.001);
+            var secondMass = firstMass.ToUnit(MassUnit.Microgram);
+
+            Assert.Equal(firstMass, secondMass);
+            Assert.Equal(secondMass, firstMass);
+        }
+
+        [Fact]
+        public void Equals_WithVeryCloseQuantityInSameUnit_ReturnsFalse()
+        {
+            var firstMass = Mass.FromGrams(0);
+            var secondMass = Mass.FromGrams(double.Epsilon);
+
+            Assert.NotEqual(firstMass, secondMass);
+            Assert.NotEqual(secondMass, firstMass);
+        }
+
+        [Fact]
+        public void GetHasCode_WithSameQuantityInAnotherUnit_ReturnsSameValue()
+        {
+            var firstMass = Mass.FromGrams(0.001);
+            var secondMass = firstMass.ToUnit(MassUnit.Microgram);
+
+            Assert.Equal(firstMass.GetHashCode(), secondMass.GetHashCode());
+        }
+
+        [Fact]
+        public void GetHasCode_WithVeryCloseQuantities_ReturnsSameValue()
+        {
+            var firstMass = Mass.FromGrams(0);
+            var secondMass = Mass.FromGrams(double.Epsilon);
+
+            Assert.Equal(firstMass.GetHashCode(), secondMass.GetHashCode());
```

Comment:
I believe you mentioned this earlier from the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=netcore-3.1#notes-to-inheritors)

> If two objects compare as equal, the GetHashCode() method for each object must return the same value. However, if two objects do not compare as equal, the GetHashCode() methods for the two objects do not have to return different values.

Reply:
They don't have to, but it can save a lot of equality checking.

```csharp
double v1 = 0.0;
double v2 = double.Epsilon;
		
Console.WriteLine(v1.GetHashCode());
Console.WriteLine(v2.GetHashCode());
```
```
0
1
```